File: leetcode/ipo/solution_test.cpp
#include "../../tools/leetcode.h"
#include "solution.cpp"

int main() {
    Solution sol;
    {
        vector<int> p = {1, 2, 3}, c = {0, 1, 1};
        test(sol.findMaximizedCapital(2, 0, p, c) == 4, "Example 1");
    }
    {
        vector<int> p = {1, 2, 3}, c = {0, 1, 2};
        test(sol.findMaximizedCapital(3, 0, p, c) == 6, "Example 2");
    }
    return summary();
}

File: leetcode/maximum-building-height/solution_test.py
import sys, os
sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))
from solution import Solution

_pass = _fail = 0

def test(result, expected, name):
    global _pass, _fail
    ok = result == expected
    print(f"{'PASS' if ok else 'FAIL'}: {name}")
    if not ok:
        print(f"  Expected: {expected!r}")
        print(f"  Got:      {result!r}")
    if ok: _pass += 1
    else:  _fail += 1

def summary():
    print(f"\n{_pass}/{_pass + _fail} tests passed")

sol = Solution()
test(sol.maxBuilding(5, [[2,1],[4,1]]), 2, "Example 1")
test(sol.maxBuilding(6, []), 5, "Example 2 - no restrictions")
test(sol.maxBuilding(10, [[5,3],[2,5],[7,4],[10,3]]), 5, "Example 3")

summary()
